site/content/extend/plugins/webapp/proper-configuration.md

```diff
@@ -0,0 +1,38 @@
+---
+title: Proper Configuration
+subsection: Web App Plugins
+weight: 0
+---
+
+### Developing in the plugin's webapp folder
+
+In order for your IDE to know the root directory of the plugin's webapp code, it is advantageous to open the IDE in the webapp folder itself when working on the webapp portion of the plugin. This way, the IDE is aware of files such as `webpack.config.js` and `tsconfig.json`.
+
+### Using the Mattermost server's SiteURL in your webapp plugin
+
+In order to make sure your plugin has full compatibility with your Mattermost server, you should use the server's configured SiteURL in each API call you send to the server from the webapp. Here's an [example](https://github.com/mattermost/mattermost-plugin-jira/blob/19a9c2442817132b4eee5c77e259b80a40188a6a/webapp/src/selectors/index.js#L13-L26) of how to compute the SiteURL:
```

Comment:
Following from above, my read on this is "web app" as it's the developer's web app?

Reply:
We usually refer to it as `webapp` when talking about the plugins and plugin repos. I'd say it's very common to do so when talking about code as well. Also, when I say "the webapp" in any of the prose in this file, I mean "the functionality the plugin is adding to the web application". It's implied that it means this, but I agree that it is not obvious.

Reply:
That makes sense. I've been trying to work out how to standardize it in the docs and your explanation works: "webapp" when referring to repos/code and "web app" when referring to "web applications" proper (as per a brief Google).